[BUG] NPM7 Fails to install: unmet peer dependency with Cypress 5+ - "could not resolve dependency"

<!-- Feel free to remove sections that aren't relevant.

## Title line template: [Title]: Brief description

-->

## What tool(s) are you using?

<!--
  Add an x in one of the options below, for example:
- [x] package name
-->

- [ ] `accessibility-checker-engine`
- [ ] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [x] `other` - `cypress-accessibility-checker`

## Detailed description

> Node version: 16.5.0
> NPM version: 7.19.1
> Platform: MacOS

> Due to changes in NPMs package lock (lockfileversion 2), an error is thrown on `npm i` when using any version of Cypress that is not explicitly called out as a peer dependency of the `cypress-accessibility-checker` package.

## Steps to reproduce the issue

1. install the latest version of node
2. add `cypress` to your package as a dev dependency (we are on v8.3.0)
3. add `cypress-accessibility-checker` as a dev dependency (we are on v3.1.9)
4. run `npm install`


## Additional information

- our project's package.json:
    ```
    "devDependencies": {
        "@babel/core": "7.14.6",
        "@cypress/webpack-preprocessor": "5.9.1",
        "@mural/ds-manifest-builder": "0.0.0",
        "@storybook/addon-a11y": "6.3.4",
        "@storybook/addon-actions": "6.3.4",
        "@storybook/addon-essentials": "6.3.4",
        "@storybook/addon-links": "6.3.4",
        "@storybook/html": "6.3.4",
        "@storybook/react": "6.3.4",
        "@types/fs-extra": "9.0.12",
        "babel-loader": "8.2.2",
        "cypress": "8.2.0",
        "cypress-accessibility-checker": "3.1.9",
        "rimraf": "3.0.2",
        "ts-loader": "8.3.0"
    },
    "dependencies": {
        "@mural/ds-component-button": "0.0.0",
        "@mural/component-starter": "0.0.0",
        "@mural/ds-config": "0.0.0"
    }
    ```
- error output in terminal:
    ```
    npm ERR! Found: cypress@8.2.0
    npm ERR! node_modules/cypress
    npm ERR!   cypress@"8.2.0" from the root project
    npm ERR!
    npm ERR! Could not resolve dependency:
    npm ERR! peer cypress@"^3 || ^4" from cypress-accessibility-checker@3.1.9
    npm ERR! node_modules/cypress-accessibility-checker
    npm ERR!   cypress-accessibility-checker@"3.1.9" from the root project
    npm ERR!
    npm ERR! Fix the upstream dependency conflict, or retry
    npm ERR! this command with --force, or --legacy-peer-deps
    npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
    npm ERR!
    ```

### Temporary Solution:
The can be fixed by running `npm install --legacy-peer-deps` - but this may require that other peer dependencies need to be added to your package to meet all requirements of other packages.

### Solving this Bug:
This bug can be solved by ensuring that the `peerDependencies` object of the `cypress-accessibility-checker` package is either maintained with new releases of cypress _or_ removed.

--- a/cypress-accessibility-checker/package.json
+++ b/cypress-accessibility-checker/package.json
@@ -30,7 +30,7 @@
   ],
   "license": "SEE LICENSE IN LICENSE from the 'equal-access' repository",
   "peerDependencies": {
-    "cypress": "^3 || ^4"
+    "cypress": "^3 || ^4 || ^5 || ^6 || ^7 || ^8"
   },
   "devDependencies": {
     "cypress": "^5.2.0",

